mono/mini/mini-posix.c

```diff
@@ -908,7 +908,7 @@ dump_memory_around_ip (MonoContext *mctx)
 static void
 print_process_map (void)
 {
-#ifdef __linux__
+#if defined(__linux__) && !defined(HOST_ANDROID)
 	FILE *fp = fopen ("/proc/self/maps", "r");
```

Comment:
It could be open/read/close instead of fopen/fgets/fclose, right?
You just have to do the newline scanning and it is pretty easy.
You don't have to output a line at a time, you just count the newlines and truncate the last output when you saw enough.